client-course-schedulizer/src/utilities/hooks/addSectionHooks.ts

```diff
@@ -61,16 +84,60 @@ export const useAddSectionToSchedule = () => {
     handleOldSection(oldData, newSection, removeOldSection, schedule);
     insertSectionCourse(schedule, newSection, newCourse);
     appDispatch({ payload: { schedule }, type: "setScheduleData" });
-    setIsCSVLoading(false);
   };
 
   return { addNonTeachingLoadToSchedule, addSectionToSchedule };
 };
 
+const switchToCorrectTerm = async (
+  newSection: Section,
+  currentTerm: Term,
+  appDispatch: React.Dispatch<AppAction> | (() => void),
+) => {
+  const newTerm = Array.isArray(newSection.term) ? newSection.term[0] : newSection.term;
+  if (newTerm !== currentTerm) {
+    await appDispatch({
+      payload: { selectedTerm: newTerm },
+      type: "setSelectedTerm",
+    });
+  }
+};
+
+const scrollToUpdatedSection = (newCourse: Course, newSection: Section) => {
+  let className = "";
+  let newElement: Element | undefined;
+  const newSectionName = `${newCourse.prefixes[0]}-${newCourse.number}-${newSection.letter}`;
+  forEach(newSection.instructors, (prof) => {
+    forEach(newSection.meetings, (meeting) => {
+      const room = `${meeting.location.building}_${meeting.location.roomNumber}`;
+      className = createEventClassName(newSectionName, room, prof);
+      const newElements = document.getElementsByClassName(className);
+      if (newElements) {
+        newElement = newElements.item(newElements.length - 1) ?? undefined;
+      }
+    });
+  });
+  if (newElement && newElement.parentElement?.parentElement) {
+    newElement.scrollIntoView({ behavior: "smooth", inline: "nearest" });
```

Comment:
I tried changing this to `inline: "center"` and it seemed to work as expected. What graphical issues did you experience?

Reply:
So to test the graphical issues, set it to `inline: "center"` like you did, import the math schedule (or another bigger schedule), and edit a section for the last teacher listed (rightmost). When it scrolls over to that section once updated, it ends up shifting all of the content of the site to the left leaving some blank space on the right and hiding some of the content on the left.

Are you able to reproduce this behavior? (I was testing on Chrome in case that's relevant)

Reply:
Yes, I see the issue now. Let's just keep it as is then.